MarkBind/vue-strap
# Migrate body docs [WIP]

**Notice:** Documentation for components are being migrated to `markbind` repository, which will render this PR obsolete. MarkBind/markbind#274

What is the purpose of this pull request? (put "X" next to an item, remove the rest)

• [X] Other, migration






What changes did you make? (Give an overview)

**Note:**
**1. const MH =  'Migration Helper';**
**2. This PR was forked from `migrate-question` to test out rendering of one component doc, will have to rebase in the near future**
**3. This [WIP] PR serves as a skeleton for team members to test migration of their component documents.**

File | Changes
--- | ---
index.html | replaced `<body-docs></body-docs>` with  `<div id="component-body">`
bodyDocs.vue | 1. Commented out or removed documents not being tested.<br>2. Corrected spelling of `compoennts` 
docs/index.js | 1. Changed file path for imports (seemed wrong)<br>2. Commented out reference to `Vue.config.debug` (**MH**)<br>3. changed `el` attribute to `#component-body` and added `template` for `<body-docs>`
src/index.js | Commented out everything unrelated to `Question.vue`
docCode.vue<br>docTable.vue | Attempted patch with **MH** suggested changes. However has console errors when rendering `bodyDocs`

Testing instructions:
1. Checkout `migrate-bodyDocs` branch.
2. Execute `npm run builddocs` then `npm run docs` in the repository terminal
3. Go to `http://localhost:8080/` and you should see the vue-strap component guide rendered for `questionDocs.vue`.<br>(Although `docTable` and `docCode` will throw console errors + not be rendered properly)

Testing your component docs:
1. Insert your doc component inside `bodyDocs.vue`'s **template** and **script** 

```
<template>
.....
    <div ref="sections" class="col-md-9">
      <question-docs></question-docs>
      <your-doc-in-kebab-case-here></your-doc-in-kebab-case-here>
    </div>
.....
</template>

<script>
import yourDocInCamelCaseHere from ./example/yourDocs.vue
.....
</script>
```

2. Go to `src/index.js` and un-comment the relevant `imports` and `const components` for your docs.
3. Repeat step 2 and 3 in **Testing instructions**
